### ISSUE TITLE
~/.ghost/config not pupulated after ghost install fron non-root user

### ISSUE THREAD

**Issue by PaszaVonPomiot** [TryGhost/Ghost-CLI#220]
There is also no info about modifying `~/.ghost/config` file when doing verbose installation.
It works however if I do the installation from root. How do I troubleshoot that?

```
[ghost@test strefadysk_prd]$ ghost ls
┌──────┬──────────┬────────┬─────┬──────┬─────────────────┐
│ Name │ Location │ Status │ Url │ Port │ Process Manager │
└──────┴──────────┴────────┴─────┴──────┴─────────────────┘
[ghost@test strefadysk_prd]$ ghost install --no-stack --pname strefadysk_prd --url 'http://test.pl'
 ✔ Checking for latest Ghost version
 ✔ Running system checks
 ✔ Setting up install directory
 ✔ Downloading and installing Ghost v1.0.0-beta.1
 ✔ Moving files
? Enter your blog URL: http://test.pl
? Enter your MySQL hostname: localhost
? Enter your MySQL username: ghost
? Enter your MySQL password: [hidden]
? Enter your Ghost database name: strefadysk_prd
✔ Finishing setup
? Do you want to start Ghost? No
[ghost@test strefadysk_prd]$ ghost ls
┌──────┬──────────┬────────┬─────┬──────┬─────────────────┐
│ Name │ Location │ Status │ Url │ Port │ Process Manager │
└──────┴──────────┴────────┴─────┴──────┴─────────────────┘
[ghost@test strefadysk_prd]$
```

```
[ghost@test strefadysk_prd]$ ls -la  ~/.ghost/
drwxrwxr-x. 2 ghost ghost 4096 Jun 12 02:53 .
drwx------. 6 ghost ghost 4096 Jun 21 15:09 ..
-rw-rw-r--. 1 ghost ghost   17 Jun 21 16:46 config

[ghost@test strefadysk_prd]$ cat ~/.ghost/config
{"instances":[]}
```

Ghost Version: beta.1
OS: Centos 7.3
Database: MariaDB

**Comment by acburdine** [comment]
Currently, it will only show up in `ghost ls` after it's been started once (that will change). However, I'm not sure what you mean by "it works from root". When you installed it as root, did you choose to start ghost during setup?

**Comment by PaszaVonPomiot** [comment]
Hmm, I can't replicate the good scenario with root anymore. 
```
ghost install

...

✔ Finishing setup
? Do you want to start Ghost? Yes
 ✔ Validating config
✔ Starting Ghost
You can access your blog at http://test.pl
[root@test strefadysk_prd]# ghost ls
┌──────┬──────────┬────────┬─────┬──────┬─────────────────┐
│ Name │ Location │ Status │ Url │ Port │ Process Manager │
└──────┴──────────┴────────┴─────┴──────┴─────────────────┘
[root@test strefadysk_prd]#
```

**Comment by acburdine** [comment]
Okay, definitely seems like a bug then. Will keep this in mind as #207 gets worked on.

**Comment by PaszaVonPomiot** [comment]
OK, I did some further testing and conclusions are:

- it doesn't matter if user is root or not
- it's not necessary to start ghost at the end of installation for `ghost ls` to show newly installed instance
- if you uninstall ghost and leave `~/.ghost/config` the next installation will not show up there
- if you manually delete `~/.ghost/config` and then `ghost install` then it will work properly

**Comment by acburdine** [comment]
Oh....that makes more sense then! Must have introduced this bug when working on cleaning up the instance list after `ghost uninstall` :(

**Comment by acburdine** [comment]
So I did figure out what this was - if you have only one instance `ghost uninstall` sets the instances parameter in `~/.ghost/config` to an empty array, rather than an empty object like it's supposed to be. The next time you run ghost setup, it tries to set a value on an object and gets an empty array so it doesn't work right.

Fix will be released in 1.0.0-alpha.19
